src/integer_mod_q/mat_zq/sample.rs

```diff
@@ -0,0 +1,179 @@
+    /// - `modulus`: specifies the modulus of the matrix and defines the interval
+    /// over which is sampled
+    ///
+    /// Returns a new [`MatZq`] instance with entries chosen
+    /// uniformly at random in `[0, modulus)` or a [`MathError`]
+    /// if the dimensions of the matrix or the modulus were chosen too small.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::MatZq;
+    ///
+    /// let matrix = MatZq::sample_uniform(3, 3, &17).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type
+    /// [`InvalidMatrix`](MathError::InvalidMatrix)
+    /// if the number of rows or columns is `0`.
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds)
+    /// if the number of rows or columns is negative or it does not fit into an [`i64`].
+    /// - Returns a [`MathError`] of type [`InvalidInterval`](MathError::InvalidInterval)
+    /// if the given `modulus` is smaller than or equal to `1`.
+    pub fn sample_uniform<T>(
+        num_rows: impl TryInto<i64> + Display + Copy,
+        num_cols: impl TryInto<i64> + Display + Copy,
+        modulus: &T,
+    ) -> Result<Self, MathError>
+    where
+        T: Into<Z> + Clone,
+    {
+        let modulus: Z = modulus.clone().into();
+        let num_rows = evaluate_index(num_rows)?;
+        let num_cols = evaluate_index(num_cols)?;
+        let mut matrix = MatZq::new(num_rows, num_cols, modulus.clone())?;
+
+        for row in 0..num_rows {
+            for col in 0..num_cols {
+                let sample = sample_uniform_rejection(&modulus)?;
+                matrix.set_entry(row, col, sample).unwrap();
+            }
+        }
+
+        Ok(matrix)
+    }
+}
+
+#[cfg(test)]
+mod test_sample_uniform {
+    use crate::traits::{GetEntry, GetNumColumns, GetNumRows};
+    use crate::{
+        integer::Z,
+        integer_mod_q::{MatZq, Modulus, Zq},
+    };
+    use std::str::FromStr;
+
+    /// Checks whether the boundaries of the interval are kept for small moduli.
+    #[test]
+    fn boundaries_kept_small() {
+        for _ in 0..32 {
+            let matrix = MatZq::sample_uniform(1, 1, &17).unwrap();
+            let sample = matrix.get_entry(0, 0).unwrap();
+            assert!(Z::ZERO <= sample);
+            assert!(sample < Z::from(17));
+        }
+    }
+
+    /// Checks whether the boundaries of the interval are kept for large moduli.
+    #[test]
+    fn boundaries_kept_large() {
+        let modulus = Z::from(u64::MAX);
+        for _ in 0..256 {
+            let matrix = MatZq::sample_uniform(1, 1, &modulus).unwrap();
+            let sample = matrix.get_entry(0, 0).unwrap();
+            assert!(Z::ZERO <= sample);
+            assert!(sample < modulus);
+        }
+    }
+
+    /// Checks whether providing an invalid interval/ modulus results in an error.
+    #[test]
+    fn invalid_interval() {
+        let mat_0 = MatZq::sample_uniform(3, 3, &-1);
```

Comment:
I See, that you have added a test for a negative modulus, but I would like one with a dedicated, and larger modulus like i64::MIN